### ISSUE TITLE
Join is pick-like, but probably should be cut-like

### ISSUE THREAD

**Issue by philrz** [brimdata/zed#2833]
Repro is with Zed commit c2a1d58.

We've discussed this one in the past, and I'm revisiting it while trying to write proper Zed docs for `join`. Seen through the eyes of the user, the current behavior feels like it could create confusion, so I'm advocating we flip it.

The first input source:

```
$ cat fruit.ndjson 
{"name":"apple","color":"red","flavor":"tart"}
{"name":"banana","color":"yellow","flavor":"sweet"}
{"name":"avocado","color":"green","flavor":"savory"}
{"name":"strawberry","color":"red","flavor":"sweet"}
{"name":"dates","color":"brown","flavor":"sweet","note":"in season"}
{"name":"figs","color":"brown","flavor":"plain"}
```

The second input source:

```
$ cat people.ndjson 
{"name":"morgan","age":61,"likes":"tart"}
{"name":"quinn","age":14,"likes":"sweet"}
{"name":"jessie","age":30,"likes":"plain"}
{"name":"chris","age":47,"likes":"tart"}
```

If I don't care about the contents of the `note` field, this initial right join works as I'd expect.

```
$ zq -version
Version: v0.29.0-402-gc2a1d587

$ cat right-join.zed 
from (
  file fruit.ndjson => sort flavor;
  file people.ndjson => sort likes;
) | right join on flavor=likes fruit:=name

$ zq -z -I right-join.zed 
{name:"jessie",age:30,likes:"plain",fruit:"figs"}
{name:"quinn",age:14,likes:"sweet",fruit:"banana"}
{name:"quinn",age:14,likes:"sweet",fruit:"strawberry"}
{name:"quinn",age:14,likes:"sweet",fruit:"dates"}
{name:"morgan",age:61,likes:"tart",fruit:"apple"}
{name:"chris",age:47,likes:"tart",fruit:"apple"}
```

However, once I express a desire to see the `note` field in the output as well, I still get all the matching left-hand records, but I only see the fields I asked for in my field-list when they're _all_ present. That is, the only fruit name that appears in the output is `dates` since it was in the only left-hand record that also had a `note` field.

```
$ cat right-join-with-note.zed 
from (
  file fruit.ndjson => sort flavor;
  file people.ndjson => sort likes;
) | right join on flavor=likes fruit:=name,note

$ zq -z -I right-join-with-note.zed 
{name:"jessie",age:30,likes:"plain"}
{name:"quinn",age:14,likes:"sweet"}
{name:"quinn",age:14,likes:"sweet"}
{name:"quinn",age:14,likes:"sweet",fruit:"dates",note:"in season"}
{name:"morgan",age:61,likes:"tart"}
{name:"chris",age:47,likes:"tart"}
```

As I've discussed with @mccanne in the past, this shows it's working similar to [`pick`](https://github.com/brimdata/zed/tree/main/docs/language/operators#pick) rather than [`cut`](https://github.com/brimdata/zed/tree/main/docs/language/operators#cut). Since we've already validated the usefulness of being able to include these kinds of "sometimes present" annotations from joined data sources, it seems it would be beneficial to switch over to the `cut`\-like behavior. I can't think of an optimal use case for the current `pick`\-like behavior offhand, but I'd not be opposed to keeping it around behind some kind of non-default flag.

**Comment by philrz** [comment]
The Zed docs published in #2836 cited this issue as a current limitation. When this issue is addressed, the docs should be updated.

**Comment by philrz** [comment]
Verified in Zed commit 32127eb.

Repeating the original repro steps, now our right join includes the `fruit` field in the records where it did not before.

```
$ zq -version
Version: v0.29.0-421-g32127eb7

$ zq -z -I right-join-with-note.zed
{name:"jessie",age:30,likes:"plain",fruit:"figs"}
{name:"quinn",age:14,likes:"sweet",fruit:"banana"}
{name:"quinn",age:14,likes:"sweet",fruit:"strawberry"}
{name:"quinn",age:14,likes:"sweet",fruit:"dates",note:"in season"}
{name:"morgan",age:61,likes:"tart",fruit:"apple"}
{name:"chris",age:47,likes:"tart",fruit:"apple"}
```

Also, per the comment above, the [`join` docs](https://github.com/brimdata/zed/tree/main/docs/language/operators#join) have been updated such that this issue is no longer listed as a limitation, and the example [https://github.com/brimdata/zed/tree/main/docs/language/operators#example-7---embedding-the-entire-opposite-record](https://github.com/brimdata/zed/tree/main/docs/language/operators#example-7---embedding-the-entire-opposite-record) that was previously framed as a workaround for this issue has been morphed into a workaround for #2815, which remains open.

Thanks @mattnibs!
